### ISSUE TITLE
[java] Enhance rule ReplaceEnumerationWithIterator

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2891]
**Is your feature request related to a problem? Please describe.**
The rule [ReplaceEnumerationWithIterator](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#replaceenumerationwithiterator) must flag all instances of Enumerator so that they use the new class Iterator instead.

**Describe the solution you'd like**
Replace all instances of Enumeration with Iterator using the method [Enumeration.asIterator introduced since Java 9](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Enumeration.html#asIterator()).
```java
package pmdtests;

import java.util.Collections;
import java.util.Enumeration;

public enum TestEnumeration {
  ;

  public static void main(String... args) {
    Enumeration<Integer> numeration = Collections.emptyEnumeration();
  }
}

```

**Describe alternatives you've considered**
None.

**Additional context**
None.
